tests/unit_tests/test_async_subtensor.py

```diff
@@ -0,0 +1,291 @@
+        return_value=b""
+    )
+
+    call_definition = {
+        "params": [
+            {"name": "coldkey", "type": "Vec<u8>"},
+            {"name": "uid", "type": "u16"},
+        ]
+    }
+    params = ["coldkey", "uid"]
+
+    # Call
+    decoded_params = await subtensor.encode_params(
+        call_definition=call_definition, params=params
+    )
+
+    # Asserts
+    subtensor.substrate.create_scale_object.call_args(
+        mocker.call("coldkey"),
+        mocker.call("Vec<u8>"),
+        mocker.call("uid"),
+        mocker.call("u16"),
+    )
+    assert decoded_params == "0x"
+
+
+@pytest.mark.asyncio
+async def test_encode_params_raises_error(subtensor, mocker):
+    """Tests encode_params with raised error."""
+    # Preps
+    subtensor.substrate.create_scale_object = mocker.AsyncMock(
+        autospec=async_subtensor.AsyncSubstrateInterface.create_scale_object
+    )
+    subtensor.substrate.create_scale_object.return_value.encode = mocker.Mock(
+        return_value=b""
+    )
+
+    call_definition = {
+        "params": [
+            {"name": "coldkey", "type": "Vec<u8>"},
+        ]
+    }
+    params = {"undefined param": "some value"}
+
+    # Call and assert
+    with pytest.raises(ValueError):
+        await subtensor.encode_params(call_definition=call_definition, params=params)
+
+        subtensor.substrate.create_scale_object.return_value.encode.assert_not_called()
+
+
+@pytest.mark.asyncio
+async def test_get_current_block(subtensor):
+    """Tests get_current_block method."""
+    # Call
+    result = await subtensor.get_current_block()
+
+    # Asserts
+    subtensor.substrate.get_block_number.assert_called_once()
+    assert result == subtensor.substrate.get_block_number.return_value
+
+
+@pytest.mark.asyncio
+async def test_get_block_hash_without_block_id_aka_none(subtensor):
+    """Tests get_block_hash method without passed block_id."""
+    # Call
+    result = await subtensor.get_block_hash()
+
+    # Asserts
+    assert result == subtensor.substrate.get_chain_head.return_value
+
+
+@pytest.mark.asyncio
+async def test_get_block_hash_with_block_id(subtensor):
+    """Tests get_block_hash method with passed block_id."""
+    # Call
+    result = await subtensor.get_block_hash(block_id=1)
+
+    # Asserts
+    assert result == subtensor.substrate.get_block_hash.return_value
+
+
+@pytest.mark.asyncio
+async def test_is_hotkey_registered_any(subtensor, mocker):
+    """Tests is_hotkey_registered_any method."""
+    # Preps
+    mocked_get_netuids_for_hotkey = mocker.AsyncMock(
+        return_value=[1, 2], autospec=subtensor.get_netuids_for_hotkey
+    )
+    subtensor.get_netuids_for_hotkey = mocked_get_netuids_for_hotkey
+
+    # Call
+    result = await subtensor.is_hotkey_registered_any(
+        hotkey_ss58="hotkey", block_hash="FAKE_HASH"
+    )
+
+    # Asserts
+    assert result == (len(mocked_get_netuids_for_hotkey.return_value) > 0)
```

Comment:
```suggestion
    assert result is (len(mocked_get_netuids_for_hotkey.return_value) > 0)
```

Use `is` for bool comparison